document/core/exec/instructions.rst

```diff
@@ -2625,16 +2625,41 @@ Control Instructions
 :math:`\TRY~\blocktype~\instr_1^\ast~(\CATCH~x~\instr_2^\ast)^\ast~(\CATCHALL~\instr_3^\ast)^?~\END`
 ....................................................................................................
 
-.. todo::
-   Add prose for the |TRY| - |CATCH| - |CATCHALL| execution step.
+1. Assert: due to :ref:`validation <valid-blocktype>`, :math:`\expand_F(\blocktype)` is defined.
+
+2. Let :math:`[t_1^m] \to [t_2^n]` be the :ref:`function type <syntax-functype>` :math:`\expand_F(\blocktype)`.
+
+3. Let :math:`L` be the label whose arity is :math:`n` and whose continuation is the end of the |TRY| instruction.
+
+4. Assert: due to :ref:`validation <valid-try-catch>`, there are at least :math:`m` values on the top of the stack.
+
+5. Pop the values :math:`\val^m` from the stack.
+
+6. Let :math:`F` be the :ref:`current <exec-notation-textual>` :ref:`frame <syntax-frame>`.
+
+7. Let :math:`H` be the empty :ref:`catching exception handler <syntax-handler>` :math:`\CATCHadm`.
+
+8. For each catch clause :math:`(\CATCH~x~\instr_2^\ast)` do:
+
+   a. Let :math:`a_x` be the tag address :math:`F.\AMODULE.\MITAGS[x]`.
+
+   b. Append :math:`\{a_x~\instr_2^\ast\}` to :math:`H`.
+
+9. If there is a catch all clause :math:`(\CATCHALL~\instr_3^\ast)`, then
+
+   a. Append :math:`\{\epsilon~\instr_3^\ast\}` to :math:`H`.
+
```

Comment:
@rossberg this is the newly added loop that constructs the catching exception handler `H`. I am not sure about also calling `(\CATCH~x~\instr_2^\ast)` a "catch clause", as this is currently the name we give to a `\CAUGHTADM` administrative instruction. They directly correspond to eachother though, in the sense that `\CAUGHTADM` is a selected `\CATCH` or `\CATCHALL`, so I used the same name to descibe it in this loop. WDYT?

Reply:
There was also [a suggestion](https://github.com/WebAssembly/exception-handling/pull/224#discussion_r972772672) by @aheejin to use "catch bodies", although IIUC this was originally meant to be used for the parts of the handler, so for `\{a_x~\instr_2^\ast\}` and `\{\epsilon~\instr_3^\ast\}`.

Not sure what's best for this version of the prose.